src/v0/destinations/optimizely_fullstack/transform.js

```diff
@@ -0,0 +1,90 @@
+const { EventType } = require('../../../constants');
+const {
+  defaultRequestConfig,
+  simpleProcessRouterDest,
+  constructPayload,
+  removeUndefinedAndNullValues,
+  defaultPostRequestConfig,
+} = require('../../util');
+const { validateConfig } = require('./util');
+const { CONFIG_CATEGORIES, MAPPING_CONFIG, getTrackEndPoint } = require('./config');
+const { TransformationError, InstrumentationError } = require('../../util/errorTypes');
+
+const responseBuilder = (payload, endpoint, destination) => {
+  if (payload) {
+    const response = defaultRequestConfig();
+    const { sdkKey } = destination.Config;
+    response.endpoint = endpoint;
+    response.headers = {
+      'Content-Type': 'application/json',
+      'X-Optimizely-SDK-Key': sdkKey,
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+    return response;
+  }
+  // fail-safety for developer error
+  throw new TransformationError('Something went wrong while constructing the payload');
+};
+
+const trackResponseBuilder = (message, destination) => {
+  const { event, anonymousId } = message;
+  if (!event) {
+    throw new InstrumentationError('Event name is required');
+  }
+  const { baseUrl, trackKnownUsers } = destination.Config;
+  const endpoint = getTrackEndPoint(baseUrl, event);
+  const { name } = CONFIG_CATEGORIES.TRACK;
+  const payload = constructPayload(message, MAPPING_CONFIG[name]);
+  if (!trackKnownUsers) {
+    payload.userId = anonymousId;
+  }
+  return responseBuilder(payload, endpoint, destination);
+};
+
+const pageResponseBuilder = (message, destination) => {
+  const { trackCategorizedPages, trackNamedPages } = destination.Config;
+  const newMessage = { ...message };
+  const returnValue = [];
+
+  if (message.category && trackCategorizedPages) {
+    newMessage.event = `Viewed ${message.category} page`;
+    returnValue.push(trackResponseBuilder(newMessage, destination));
+  }
+  if (message.name && trackNamedPages) {
+    newMessage.event = `Viewed ${message.name} page`;
+    returnValue.push(trackResponseBuilder(newMessage, destination));
+  }
```

Comment:
what happens if both name and category are present? and also if both `trackCategorizedPages` and `trackNamedPages` are turned on?